devicetypes/smartthings/zigbee-contact-sensor.src/zigbee-contact-sensor.groovy

```diff
@@ -0,0 +1,207 @@
+		capability "Battery"
+		capability "Configuration"
+		capability "Contact Sensor"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Sensor"
+
+		fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000, 0003, 0500", manufacturer: "LUMI", model: "lumi.magnet.agl02", deviceJoinName: "LUMI Contact Sensor"
+	}
+
+	simulator {
+
+		status "open": "zone status 0x0021 -- extended status 0x00"
+		status "close": "zone status 0x0000 -- extended status 0x00"
+
+		for (int i = 0; i <= 90; i += 10) {
+			status "battery 0x${i}": "read attr - raw: 2E6D01000108210020C8, dni: 2E6D, endpoint: 01, cluster: 0001, size: 08, attrId: 0021, encoding: 20, value: ${i}"
+		}
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name: "contact", type: "generic", width: 6, height: 4) {
+			tileAttribute("device.contact", key: "PRIMARY_CONTROL") {
+				attributeState "open", label: '${name}', icon: "st.contact.contact.open", backgroundColor: "#e86d13"
+				attributeState "closed", label: '${name}', icon: "st.contact.contact.closed", backgroundColor: "#00A0DC"
+			}
+		}
+
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main(["contact"])
+		details(["contact", "battery", "refresh"])
+	}
+}
+
+def parse(String description) {
+	Map map = [:]
+
+	List listMap = []
+	List listResult = []
+
+	log.debug "parse: Parse message: ${description}"
+
+	if (description?.startsWith("enroll request")) {
+		List cmds = zigbee.enrollResponse()
+
+		log.debug "parse: enrollResponse() ${cmds}"
+		listResult = cmds?.collect { new physicalgraph.device.HubAction(it) }
+	}
+	else {
+		if (description?.startsWith("zone status")) {
+			listMap = parseIasMessage(description)
+		}
+		else if (description?.startsWith("catchall:")) {
+			map = parseCatchAllMessage(description)
+		}
+
+		// Create events from map or list of maps, whichever was returned
+		if (listMap) {
+			for (msg in listMap) {
+				listResult << createEvent(msg)
+			}
+		}
+		else if (map) {
+			listResult << createEvent(map)
+		}
+	}
+
+	log.debug "parse: listResult ${listResult}"
+	return listResult
+}
+
+private Map parseCatchAllMessage(String description) {
+	Map resultMap = [:]
+	def cluster = zigbee.parse(description)
+
+	if (shouldProcessMessage(cluster)) {
+		def msgStatus = cluster.data[2]
+
+		log.debug "parseCatchAllMessage: msgStatus: ${msgStatus}"
+		if (msgStatus == 0) {
+			switch(cluster.clusterId) {
+				case 0x0500:
+                	Map descMap = zigbee.parseDescriptionAsMap(description)
```

Comment:
You have some indentation issues, you should make sure that you are consistently using tabs.